### PR TITLE
APE8: Astropy Community Code of Conduct

### PR DESCRIPTION
**Pull request by eteq**
This APE proposes a code of conduct for the astropy community.  Details and text of the code are in the APE8.rst file.

For a more detailed edit history, you can look at the https://docs.google.com/document/d/1yLObQMqP66x6MNuHXFpW_4dLPBtNp--9BHxfG8YdPIY google doc.

### CODE REVIEW

**Comment by crawfordsm** [comment]
Should something be added to the 'Implementation' that a copy of the code of conduct should be included in the repositories for the code (affiliated packages) and not just on the website?   Or should the packages just have a link referencing the website with the Code of Conduct?

**Comment by embray** [comment]
:+1: In principle to including a copy in the code.  :-1: in practice due to difficulty of profliferating changes/updates.  :+1: to a link though, in the readme.

**Comment by astrofrog** [comment]
I think it should be on the website. The code of conduct can't wait for releases to be updated :) :+1: to a link, maybe in the CONTRIBUTING.md file when available?

File: APE8.rst [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+The text of the code of conduct is shown here:
+
+Astropy Community Code of Conduct
+=================================
+
+The community of participants in open source Astronomy projects is made up of members from around the globe with a diverse set of skills, personalities, and experiences. It is through these differences that our community experiences success and continued growth. We expect everyone in our community to follow these guidelines when interacting with others both inside and outside of our community. Our goal is to keep ours a positive, inclusive, successful, and growing community.
+
+As members of the community,
+
+* We pledge to treat all people with respect and provide a harassment- and bullying-free environment, regardless of sex, sexual orientation and/or gender identity, disability, physical appearance, body size, race, nationality, ethnicity, and religion. In particular, sexual language and imagery, sexist, racist,  or exclusionary jokes are not appropriate.
```
Comment (embray) [inline APE8.rst]: Extra space between "racist, or".    I wonder if it would make sense to say "otherwise exclusionary" given that the previously listed items are already exclusionary by their nature.  On the other hand maybe that's implicit and I'm being characteristically too wordy.

File: APE8.rst [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+* We pledge to respect the work of others by recognizing acknowledgment/citation requests of original authors. As authors, we pledge to be explicit about how we want our own work to be cited or acknowledged.
+* We pledge to welcome those interested in joining the community, and realize that including people with a variety of opinions will only serve to enrich our community. In particular, discussions relating to pros/cons of various technologies, programming languages, and so on are welcome, but these should be done with respect, taking proactive measure to ensure that all participants are heard and feel confident that they can freely express their opinions.
```
Comment (embray) [inline APE8.rst]: How about "variety of opinions *and* backgrounds" ?

File: APE8.rst [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+* We pledge to welcome questions and answer them respectfully, paying particular attention to those new to the community. We pledge to provide respectful criticisms and feedback in forums, especially in discussion threads resulting from code contributions.
+* We pledge to be conscientious of the perceptions of the wider community and to respond to criticism respectfully. We will strive to model behaviors that encourage productive debate and disagreement, both within our community and where we are criticized. We will treat those outside our community with the same respect as people within our community.
+* We pledge to help the entire community follow the code of conduct, and to not remain silent when we see violations of the code of conduct. We will take action when members of our community violate this code such as contacting community@astropy.org or talking privately with the person.
```
Comment (embray) [inline APE8.rst]: I feel like this line still needs a bit of kneading, though I know it's already been discussed a fair bit, and I'm sorry if I lost track of that discussion after the last week.  Two issues with this:

1) Although *we* may feel confident that whoever is behind community@astropy.org will handle situations swiftly and appropriately, an outside reader may not share that confidence.  It's not clear from this document alone who is behind that e-mail address or why they should be contacted.  I guess the CoC itself pledges that "we will take action", but a single e-mail address still may appear as a black box. 

2) Related to that, "talking privately" may not always be an option.  See for example recent discussions about anonymous reviewers, and slowness of publishers to respond to issues with reviewers. Not 100% sure what the best solution is.  Maybe more information about community@astropy.org somewhere below would help.

**Comment by eteq** [comment]
(In the description of this PR I should have @-mentioned the other authors: @astrofrog @kelle @alexavillaume @astronomeralex)

File: APE8.rst (hunk 3 above)
Comment (astrofrog) [inline APE8.rst]: Good point about the email address. How about actually having an email address called confidential@astropy.org and then also say in parentheses (all emails sent to this address will be treated with the strictest confidence)

Comment (embray) [inline APE8.rst]: Though maybe also expediently.

**Comment by eteq** [comment]
I've addressed @embray's suggestions, although note I took @astrofrog's wording *without* the use of "expedient": I'm a bit concerned about adding that because it's pretty unclear exactly what timescale that means but also don't want to give an exact timeline given the problem of "what if coordination committee members are traveling" and so on.

**Comment by eteq** [comment]
And if anyone's worried, I've already set up the astropy.org domain to send on e-mails from these addresses

**Comment by astrofrog** [comment]
@eteq @perrygreenfield - shall we go ahead and approve this?

**Comment by perrygreenfield** [comment]
yes on my part

File: APE8.rst [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+* We pledge to help the entire community follow the code of conduct, and to not remain silent when we see violations of the code of conduct. We will take action when members of our community violate this code such as contacting confidential@astropy.org (all emails sent to this address will be treated with the strictest confidence) or talking privately with the person.
+
+This code of conduct applies to all community situations online and offline, including mailing lists, forums, social media, conferences, meetings, associated social events, and one-to-one interactions.
+
+Parts of this code of conduct have been adapted from the `PSF code of conduct <https://www.python.org/psf/codeofconduct/>`_.
+
+The Astropy Community Code of Conduct is licensed under a
+Creative Commons Attribution 4.0 International License.
+You should have received a copy of the license along with this
+work. If not, see http://creativecommons.org/licenses/by/4.0/.
```
Comment (astrofrog) [inline APE8.rst]: Maybe just:

```
The Astropy Community Code of Conduct is licensed under a [Creative Commons Attribution 4.0 International License](http://creativecommons.org/licenses/by/4.0/)
```

Comment (eteq) [inline APE8.rst]: good point - I'll make that change for the plain text.  It's not really that important in the end because I'll put the official creative commons HTML code for the web site itself.

**Comment by astrofrog** [comment]
@eteq - since @perrygreenfield gave the go-ahead above, I think it's fine for you to update the decision rationale and merge this.
